### ISSUE TITLE
Header numbering no longer in output from quarto filters

### ISSUE THREAD

**Issue by matthew-brett** [quarto-dev/quarto-cli#200]
I just noticed (from my projects tests) that Quarto no longer writes out the header numbers into the documents that become the inputs to custom filters.

Checking with an old version of Quarto (v0.2.183), I see this in the input JSON for one document with a heading:

```
                {
                  "t": "Span",
                  "c": [
                    [
                      "",
                      [
                        "header-section-number"
                      ],
                      []
                    ],
                    [
                      {
                        "t": "Str",
                        "c": "2.1"
                      }
                    ]
                  ]
                },
```

but - this content is missing from the output of current Quarto, though I see the heading number does reach the output HTML.

The heading number was useful, because we are using Pandoc filters to write out notebook versions of the pages, and it's convenient to have the heading numbers in the notebook, so the reader can cross-reference with the web or PDF document.

Is there any way of getting the heading numbers now?  How do they reach the final document?

**Comment by jjallaire** [comment]
We did recently make a change to ensure that user filters run *after* the Quarto filters (which should in your case be exactly what you want as Quarto would have put in the section number). Do you have any custom ordering of filters going on in your config file?

**Comment by matthew-brett** [comment]
Right - yes - our filters run after the Quarto filters already - for example in our [filter project test book](https://github.com/matthew-brett/noteout/blob/main/noteout/tests/qbook/_quarto.yml).

**Comment by jjallaire** [comment]
I wonder if this is a pandoc difference then? (As the newer versions of
Quarto have Pandoc 2.16.1)

Could you put together an MRE and I’ll have a closer look?

**Comment by matthew-brett** [comment]
Have a look at https://github.com/matthew-brett/mysite . It has a page `a_page.qmd` with a sub-heading "An example-heading".  I added a filter [noteout-write-doc](https://github.com/matthew-brett/noteout/blob/main/noteout/write_doc.py) that just writes out the JSON for the documents it is sent.   I ran the build for Quarto v0.2.183, that generated a file  for `a_page.qmd` named `doc_001.json`, that I renamed to `a_page_v0.2.183.json`.  It has the header numbering.    Then I updated to Quarto commit `37631ad5` and ran the build again.   I renamed the `doc_003.json` corresponding to `a_page.qmd` to `a_page_37631ad5.json`.   It does not have header numbering.
